changelog/12204.bugfix.rst

```diff
@@ -0,0 +1,7 @@
+Fix a regression in pytest 8.0 where tracebacks get longer and longer when multiple tests fail due to a shared higher-scope fixture which raised.
+
+Also fix a similar regression in pytest 5.4 for collectors which raise during setup.
+
+The fix necessitated internal changes which may affect some plugins:
+- ``FixtureDef.cached_result[2]`` is now a tuple ``(exc, tb)`` instead of ``exc``.
+- ``SetupState.stack`` failures are now a tuple ``(exc, tb)`` instead of ``exc``.
```

Comment:
@bluetech FYI this list seems to need indentation or something like that. See the preview of how it's rendered: https://docs.pytest.org/en/latest/changelog.html#bug-fixes.

P.S. I integrated my changelog draft preview Sphinx extension during the sprint. So now, you're going to be able to preview the rendering in new pull requests, even before merging.

Reply:
I submitted a fix via https://github.com/pytest-dev/pytest/pull/12563.